tests/core_tests.py

```diff
@@ -551,6 +551,14 @@ def test_template_kwarg(self):
         rendered = tp.process_template(s, foo="bar")
         self.assertEqual("bar", rendered)
 
+    def test_custom_process_template(self):
+        db = mock.Mock()
+        db.backend = "presto"
+        s = "$foo()"
```

Comment:
let's have more involved test case with the function e.g. DATE or smth like that